### PR TITLE
[F1] Pass new arguments to Executor (4th)

### PR DESCRIPTION
**Pull request by FenrirUnbound**
Resolving the merge conflicts with PR #32.

[Feature](https://github.com/screwdriver-cd/screwdriver/issues/77)

Passing:
- buildId - Build Identifier (to look up stuff)
- container - Container to start in
- apiUri - URI to hit our API
- token - Token to be able to read/write from the API

Closes #32

### CODE REVIEW

File: lib/buildFactory.js [hunk 1]
```diff
@@ -93,14 +95,16 @@ class BuildFactory extends BaseFactory {
...
+
         return getCommitSha(this.datastore, modelConfig)
             .then(sha => {
                 modelConfig.sha = sha;
 
                 return super.create(modelConfig);
             })
             .then(build =>
-                build.start()
+                build.start({ apiUri })
```
Comment (nkatzman) [inline lib/buildFactory.js]: does this need to pass in `tokenGen` ?

File: lib/buildFactory.js [hunk 2]
```diff
@@ -93,14 +95,16 @@ class BuildFactory extends BaseFactory {
             status: 'QUEUED'
         }, config);
 
+        delete modelConfig.apiUri;
+
         return getCommitSha(this.datastore, modelConfig)
             .then(sha => {
                 modelConfig.sha = sha;
 
                 return super.create(modelConfig);
```
Comment (d2lam) [inline lib/buildFactory.js:109]: modelConfig includes the `username` and `executor`, I don't they they should be passed to `create`.

File: lib/buildFactory.js [hunk 3]
```diff
@@ -93,14 +95,16 @@ class BuildFactory extends BaseFactory {
             status: 'QUEUED'
         }, config);
 
+        delete modelConfig.apiUri;
```
Comment (d2lam) [inline lib/buildFactory.js]: Probably better to remove this line and add 
`hoek.applyToDefaults({status: 'QUEUED'}, { sha })` to line 103

File: lib/build.js [hunk 4]
```diff
@@ -20,6 +20,7 @@ class BuildModel extends BaseModel {
      */
     constructor(config) {
         super('build', config);
+        this.apiUri = config.apiUri;
```
Comment (stjohnjohnson) [inline lib/build.js]: lingering

**Comment by stjohnjohnson** [comment]
LGTM 👍 
yes

**Comment by d2lam** [comment]
My comments still stand...

File: lib/buildFactory.js (hunk 2 above)
Comment (FenrirUnbound) [inline lib/buildFactory.js:109]: Update in PR #42 provides a better solution for this

**Comment by FenrirUnbound** [comment]
> @d2lam: My comments still stand...

I didn't forget them 😸  I made a [separate PR](https://github.com/screwdriver-cd/models/pull/42) to better filter what gets saved in a `.create()` operation. Once that gets merged, we can pull in those updates into this PR and be able to leverage that solution

**Comment by stjohnjohnson** [comment]
This should be a major version bump, the input changes.

**Comment by d2lam** [comment]
Sure. This LGTM 👍 as long as the other one works as expected. 
And yeah probably need to bump version.

**Comment by jer** [comment]
👍

**Comment by petey** [comment]
👍  
upside_down

**Pull request merged**